agent/consul/fsm/snapshot_oss.go

```diff
@@ -574,9 +575,51 @@ func restoreACL(header *SnapshotHeader, restore *state.Restore, decoder *codec.D
 	return nil
 }
 
-// DEPRECATED (ACL-Legacy-Compat) - remove once v1 acl compat is removed
-func restoreACLBootstrap(header *SnapshotHeader, restore *state.Restore, decoder *codec.Decoder) error {
-	var req structs.ACLBootstrap
+// TODO(ACL-Legacy-Compat) - remove in phase 2
+type LegacyACL struct {
+	ID    string
+	Name  string
+	Type  string
+	Rules string
+
+	structs.RaftIndex
+}
+
+// TODO(ACL-Legacy-Compat): remove in phase 2, used by snapshot restore
+func (a LegacyACL) Convert() *structs.ACLToken {
+	correctedRules := structs.SanitizeLegacyACLTokenRules(a.Rules)
+	if correctedRules != "" {
+		a.Rules = correctedRules
+	}
+
+	token := &structs.ACLToken{
+		AccessorID:        "",
+		SecretID:          a.ID,
+		Description:       a.Name,
+		Policies:          nil,
+		ServiceIdentities: nil,
+		NodeIdentities:    nil,
+		Type:              a.Type,
+		Rules:             a.Rules,
+		Local:             false,
+		RaftIndex:         a.RaftIndex,
+	}
+
+	token.SetHash(true)
+	return token
+}
+
+// TODO(ACL-Legacy-Compat) - remove in phase 2
```

Comment:
Isn't ACL bootstrapping still done? why would the associated restore be removed later?

Reply:
Ah, yes it is. I thought we used a different request for the new system, but it appears we do still use this one. So I'll remove this TODO, and the other one on the request type.

Reply:
Actually, looking at this closer, I think this TODO is correct.

We use these request types in two ways: 
1. for routing FSM commands
2. for snapshot/restore

We still need   `ACLBootstrapRequestType` for the FSM, but this restore function is only for the legacy system. The new system writes to `tableIndex` on bootstrap, which is handled by 

```
registerRestorer(structs.IndexRequestType, restoreIndex)
```

So this TODO is correct, but the one above on `ACLBootstrapRequestType` is wrong. I'll remove that TODO.